### DISCUSSION TITLE
SQS timeout

### DISCUSSION THREAD

**Discussion by kodedewe** [brefphp/bref#1110] [Support]
Hi, I am using the laravel from queue example (https://github.com/brefphp/examples/tree/master/Laravel/queues).
Try to access the route but hit the timeout.

When I checked the cloudwatch:
- Web -> show timeout after 28s
- Worker -> no log

Check on the SQS but there is no messages.

I think the job function is very basic but why I still got the timeout.
for the IAM policy, already give full access to the SQS and lambda.
Pls help, thanks

**Comment by t-richard** [comment]
Seeing your previous question about RDS, I suppose your web lambda function is in a VPC with your RDS instance.

When in a VPC, a Lambda function cannot access the internet which it needs to reach the SQS API to post a message to the queue.

The timeout you're seeing is probably because you haven't configured a NAT gateway to enable internet access for your Lambda function.

See this link for more information on this: https://bref.sh/docs/environment/database.html#accessing-the-internet

**Reply by kodedewe** [reply]
OMG thanks

**Reply by kodedewe** [reply]
Hi Richard, thanks for your answer. After I created the NAT the timeout issue is gone.
But when try to send messages to SQS I got new issue, this is my error messages.

`Error executing "SendMessage" on "https://sqs.ap-southeast-1.amazonaws.com/504474178866/livelync-bref-dev-jobs"; AWS HTTP error: Client error: `POST https://sqs.ap-southeast-1.amazonaws.com/504474178866/livelync-bref-dev-jobs` resulted in a `403 Forbidden` response:
<?xml version="1.0"?><ErrorResponse xmlns="http://queue.amazonaws.com/doc/2012-11-05/"><Error><Type>Sender</Type><Code>I (truncated...)
InvalidClientTokenId (client): The security token included in the request is invalid. - <?xml version="1.0"?><ErrorResponse xmlns="http://queue.amazonaws.com/doc/2012-11-05/"><Error><Type>Sender</Type><Code>InvalidClientTokenId</Code><Message>The security token included in the request is invalid.</Message><Detail/></Error><RequestId>c438be9c-3810-5dc5-80f2-e4b62b428f21</RequestId></ErrorResponse>`

**Reply by kodedewe** [reply]
ah solved the issue. need to add 'token' => env('AWS_SESSION_TOKEN'), on config/queue.php
